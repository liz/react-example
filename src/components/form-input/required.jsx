import React from 'react';
import PropTypes from 'prop-types';

export const FormInputRequired = (props) => {
    if (props.required && !props.fieldError) {
        return <p className="required">* Required</p>;
    }

    return null;
}

FormInputRequired.defaultProps = {
    required: false,
    fieldError: ''
};

FormInputRequired.propTypes = {
    /** Is this field required? */
    required: PropTypes.bool, 
    /** Is there a fieldError supplied? */
    fieldError: PropTypes.string,
};