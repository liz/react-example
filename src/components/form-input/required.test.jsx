import React from 'react';
import { mount } from 'enzyme';

import { FormInputRequired } from './required';

describe('FormInputRequired', () => {
    it('should match the snapshot', () => {
        const wrapper = mount(<FormInputRequired required />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders FormInputRequired if required prop is true and fieldError prop is not supplied', () => {
        const wrapper = mount(<FormInputRequired required />);

        expect(wrapper.text()).toEqual('* Required');
    });

    it('does not render FormInputRequired if required prop is true but fieldError prop is supplied', () => {
        const wrapper = mount(
            <FormInputRequired fieldError="Error message example" required />
        );

        expect(wrapper.html()).toBeNull();
    });

    it('does not render FormInputRequired if required prop is false', () => {
        const wrapper = mount(<FormInputRequired />);

        expect(wrapper.html()).toBeNull();
    });
});