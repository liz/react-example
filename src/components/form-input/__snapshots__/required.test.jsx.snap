// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputRequired should match the snapshot 1`] = `"<p class=\\"required\\">* Required</p>"`;
